src/proposals/mips/mip-m27/mip-m27.sol

```diff
@@ -0,0 +1,324 @@
+//SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.19;
+
+import "@forge-std/Test.sol";
+
+import {Addresses} from "@proposals/Addresses.sol";
+import {HybridProposal} from "@proposals/proposalTypes/HybridProposal.sol";
+import {MockERC20Params} from "@test/mock/MockERC20Params.sol";
+import {ParameterValidation} from "@proposals/utils/ParameterValidation.sol";
+
+/// DO_VALIDATE=true DO_DEPLOY=true DO_PRINT=true DO_BUILD=true DO_RUN=true forge script
+/// src/proposals/mips/mip-m27/mip-m27.sol:mipm27
+contract mipm27 is HybridProposal, ParameterValidation {
+    string public constant override name = "MIP-M27";
+
+    uint256 public constant NEW_MXC_USDC_RESERVE_FACTOR = 0.30e18;
+    uint256 public constant NEW_MXC_USDT_RESERVE_FACTOR = 0.30e18;
+    uint256 public constant NEW_USDCWH_RESERVE_FACTOR = 0.30e18;
+
+    uint256 public constant NEW_M_USDCWH_COLLATERAL_FACTOR = 0.59e18;
+    uint256 public constant NEW_M_WBTCWH_COLLATERAL_FACTOR = 0.32e18;
+    uint256 public constant NEW_M_ETHWH_COLLATERAL_FACTOR = 0.49e18;
+
+    constructor() {
+        bytes memory proposalDescription = abi.encodePacked(
+            vm.readFile("./src/proposals/mips/mip-m27/MIP-M27.md")
+        );
+        _setProposalDescription(proposalDescription);
+    }
+
+    /// @notice proposal's actions happen only on moonbeam
+    function primaryForkId() public view override returns (uint256) {
+        return moonbeamForkId;
+    }
+
+    function deploy(Addresses addresses, address) public override {
+        /// DO NOT ADD any mock tokens to Addresses object, just use them to etch bytecode
+
+        {
```

Comment:
Can you please tell me where these mocks are being used? I didn't get the reason to have these mocks